In [418]:
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

In [419]:
df_anim = pd.read_csv('csvfiles/animal.csv')
df_veg = pd.read_csv('csvfiles/vegetal.csv')
df_pop = pd.read_csv('csvfiles/population.csv')

In [420]:
df_anim.columns

Index(['Domain Code', 'Domain', 'Area Code', 'Area', 'Element Code', 'Element',
       'Item Code', 'Item', 'Year Code', 'Year', 'Unit', 'Value', 'Flag',
       'Flag Description'],
      dtype='object')

In [421]:
df_pop.columns

Index(['Domain Code', 'Domain', 'Area Code', 'Area', 'Element Code', 'Element',
       'Item Code', 'Item', 'Year Code', 'Year', 'Unit', 'Value', 'Flag',
       'Flag Description'],
      dtype='object')

In [422]:
df_anim.size

430150

In [423]:
df_veg.size

1268064

In [424]:
df_pop.size

4984

La taille de la dataframe animal est de 430 150 elements. (rows * columns) <br>
Celle de la dataframe vegetal est de 1 268 064 elements. <br>
Et enfin la dataframe population contient 4984 elements. <br>

In [425]:
df_anim.head(5)

Domain Code                 Domain  Area Code         Area  Element Code  \
0         FBS  Food Balances (2010-)          2  Afghanistan           645   
1         FBS  Food Balances (2010-)          2  Afghanistan           645   
2         FBS  Food Balances (2010-)          2  Afghanistan           664   
3         FBS  Food Balances (2010-)          2  Afghanistan           664   
4         FBS  Food Balances (2010-)          2  Afghanistan           674   

                                  Element  Item Code         Item  Year Code  \
0     Food supply quantity (kg/capita/yr)       2731  Bovine Meat       2018   
1     Food supply quantity (kg/capita/yr)       2731  Bovine Meat       2019   
2           Food supply (kcal/capita/day)       2731  Bovine Meat       2018   
3           Food supply (kcal/capita/day)       2731  Bovine Meat       2019   
4  Protein supply quantity (g/capita/day)       2731  Bovine Meat       2018   

   Year             Unit  Value Flag Flag Description  
0  2018               kg   2.62   Fc  Calculated data  
1  2019               kg   2.65   Fc  Calculated data  
2  2018  kcal/capita/day  15.00   Fc  Calculated data  
3  2019  kcal/capita/day  16.00   Fc  Calculated data  
4  2018     g/capita/day   1.09   Fc  Calculated data

In [426]:
df_veg.head(5)

Domain Code                 Domain  Area Code         Area  Element Code  \
0         FBS  Food Balances (2010-)          2  Afghanistan           645   
1         FBS  Food Balances (2010-)          2  Afghanistan           645   
2         FBS  Food Balances (2010-)          2  Afghanistan           664   
3         FBS  Food Balances (2010-)          2  Afghanistan           664   
4         FBS  Food Balances (2010-)          2  Afghanistan           674   

                                  Element  Item Code                Item  \
0     Food supply quantity (kg/capita/yr)       2511  Wheat and products   
1     Food supply quantity (kg/capita/yr)       2511  Wheat and products   
2           Food supply (kcal/capita/day)       2511  Wheat and products   
3           Food supply (kcal/capita/day)       2511  Wheat and products   
4  Protein supply quantity (g/capita/day)       2511  Wheat and products   

   Year Code  Year             Unit    Value Flag Flag Description  
0       2018  2018               kg   160.12   Fc  Calculated data  
1       2019  2019               kg   161.93   Fc  Calculated data  
2       2018  2018  kcal/capita/day  1372.00   Fc  Calculated data  
3       2019  2019  kcal/capita/day  1387.00   Fc  Calculated data  
4       2018  2018     g/capita/day    37.00   Fc  Calculated data

In [427]:
df_pop.head(5)

Domain Code                 Domain  Area Code         Area  Element Code  \
0         FBS  Food Balances (2010-)          2  Afghanistan           511   
1         FBS  Food Balances (2010-)          2  Afghanistan           511   
2         FBS  Food Balances (2010-)          3      Albania           511   
3         FBS  Food Balances (2010-)          3      Albania           511   
4         FBS  Food Balances (2010-)          4      Algeria           511   

                         Element  Item Code        Item  Year Code  Year  \
0  Total Population - Both sexes       2501  Population       2018  2018   
1  Total Population - Both sexes       2501  Population       2019  2019   
2  Total Population - Both sexes       2501  Population       2018  2018   
3  Total Population - Both sexes       2501  Population       2019  2019   
4  Total Population - Both sexes       2501  Population       2018  2018   

           Unit  Value Flag   Flag Description  
0  1000 persons  37172    *  Unofficial figure  
1  1000 persons  38042    *  Unofficial figure  
2  1000 persons   2883    *  Unofficial figure  
3  1000 persons   2881    *  Unofficial figure  
4  1000 persons  42228    *  Unofficial figure

<h1>Nettoyage</h1>

In [428]:
df_anim.insert(14,'type', 'animal_product', allow_duplicates=False)

In [429]:
df_anim.head()

Domain Code                 Domain  Area Code         Area  Element Code  \
0         FBS  Food Balances (2010-)          2  Afghanistan           645   
1         FBS  Food Balances (2010-)          2  Afghanistan           645   
2         FBS  Food Balances (2010-)          2  Afghanistan           664   
3         FBS  Food Balances (2010-)          2  Afghanistan           664   
4         FBS  Food Balances (2010-)          2  Afghanistan           674   

                                  Element  Item Code         Item  Year Code  \
0     Food supply quantity (kg/capita/yr)       2731  Bovine Meat       2018   
1     Food supply quantity (kg/capita/yr)       2731  Bovine Meat       2019   
2           Food supply (kcal/capita/day)       2731  Bovine Meat       2018   
3           Food supply (kcal/capita/day)       2731  Bovine Meat       2019   
4  Protein supply quantity (g/capita/day)       2731  Bovine Meat       2018   

   Year             Unit  Value Flag Flag Description            type  
0  2018               kg   2.62   Fc  Calculated data  animal_product  
1  2019               kg   2.65   Fc  Calculated data  animal_product  
2  2018  kcal/capita/day  15.00   Fc  Calculated data  animal_product  
3  2019  kcal/capita/day  16.00   Fc  Calculated data  animal_product  
4  2018     g/capita/day   1.09   Fc  Calculated data  animal_product

In [430]:
df_veg.insert(14,'type', 'vegetal_product', allow_duplicates=False)

In [431]:
df_veg.head()

Domain Code                 Domain  Area Code         Area  Element Code  \
0         FBS  Food Balances (2010-)          2  Afghanistan           645   
1         FBS  Food Balances (2010-)          2  Afghanistan           645   
2         FBS  Food Balances (2010-)          2  Afghanistan           664   
3         FBS  Food Balances (2010-)          2  Afghanistan           664   
4         FBS  Food Balances (2010-)          2  Afghanistan           674   

                                  Element  Item Code                Item  \
0     Food supply quantity (kg/capita/yr)       2511  Wheat and products   
1     Food supply quantity (kg/capita/yr)       2511  Wheat and products   
2           Food supply (kcal/capita/day)       2511  Wheat and products   
3           Food supply (kcal/capita/day)       2511  Wheat and products   
4  Protein supply quantity (g/capita/day)       2511  Wheat and products   

   Year Code  Year             Unit    Value Flag Flag Description  \
0       2018  2018               kg   160.12   Fc  Calculated data   
1       2019  2019               kg   161.93   Fc  Calculated data   
2       2018  2018  kcal/capita/day  1372.00   Fc  Calculated data   
3       2019  2019  kcal/capita/day  1387.00   Fc  Calculated data   
4       2018  2018     g/capita/day    37.00   Fc  Calculated data   

              type  
0  vegetal_product  
1  vegetal_product  
2  vegetal_product  
3  vegetal_product  
4  vegetal_product

In [432]:
df_anim.size + df_veg.size

1819515

In [433]:
product = pd.concat([df_anim, df_veg])
product.head(1)

Domain Code                 Domain  Area Code         Area  Element Code  \
0         FBS  Food Balances (2010-)          2  Afghanistan           645   

                               Element  Item Code         Item  Year Code  \
0  Food supply quantity (kg/capita/yr)       2731  Bovine Meat       2018   

   Year Unit  Value Flag Flag Description            type  
0  2018   kg   2.62   Fc  Calculated data  animal_product

In [434]:
product.tail(1)

Domain Code                 Domain  Area Code      Area  Element Code  \
90575         FBS  Food Balances (2010-)        181  Zimbabwe           684   

                                  Element  Item Code           Item  \
90575  Fat supply quantity (g/capita/day)       2899  Miscellaneous   

       Year Code  Year          Unit  Value Flag Flag Description  \
90575       2019  2019  g/capita/day   0.03   Fc  Calculated data   

                  type  
90575  vegetal_product

In [435]:
product.size

1819515